GoogleCalendar.js

```diff
@@ -0,0 +1,170 @@
+const { google } = require('googleapis');
+const config = require('./config.json');
+
+/**
+ * A brief style note: I'm using the promise variant of the Google Calendar
+ * API because I can't figure out how to get the callback version to play
+ * nicely with async/await. I'm not super attached to this style, but it
+ * works well enough for now. If I learn better ways I'll tidy it up then.
+ */
+
+module.exports = class GoogleCalendar {
+  jwtClient = null;
+  calendarApi = null;
+
+  // ACL rule to make a calendar public
+  aclRule = {
+    scope: { type: 'default' },
+    role: 'reader',
+  };
+
+  constructor() {
+    if (!config.googleApiClientEmail || !config.googleApiPrivateKey) {
+      throw new Error('This instance of AginahBot is not configured to support Google Calendar integration.');
+    }
+
+    this.jwtClient = new google.auth.JWT(
+      config.googleApiClientEmail,
+      null,
+      config.googleApiPrivateKey,
+      ['https://www.googleapis.com/auth/calendar'],
+      null,
+    );
+
+    this.calendarApi = google.calendar({ version: 'v3', auth: this.jwtClient });
+  }
+
+  /**
+   * Produces a formatted URL for a calendar
+   * @param {String} id
+   * @returns {String}
+   */
+  static createUrlFromCalendarId(id) {
+    return `https://calendar.google.com/calendar/embed?src=${id.replace('@', '%40')}`;
```

Comment:
I've run into an issue stemming from this line. When I tried to create a schedule board _after_ having enabled calendar integration and scheduling an event, I ran into this error:

```
TypeError: Cannot read properties of undefined (reading 'replace')
    at GoogleCalendar.createUrlFromCalendarId (C:\Users\Ladybunne\repos\AginahBot\GoogleCalendar.js:43:65)
    at updateScheduleBoard (C:\Users\Ladybunne\repos\AginahBot\lib.js:368:35)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.execute (C:\Users\Ladybunne\repos\AginahBot\slashCommandCategories\scheduling.js:725:11)
```

Not exactly sure what's going on here.

Reply:
> Cannot read properties of undefined (reading 'replace')

In GoogleCalendar.js on line 43 it was trying to access the `replace` property on an `undefined` variable.
The undefined variable, `id` is expected to be a string, which you can see in the docblock for the function as:
> `@param {String} id`

The trace leads you to `lib.js` line 368:
```js
value: GoogleCalendar.createUrlFromCalendarId(board.googleCalendarId),
```

There, you can see the argument used is `boards.googleCalendarId`, which is where the problem lies. The reference is to `boards`, when it should be to `board`, since we are looping over `boards`, which does not have the `googleCalendarId` property. Therefore, passing `undefined` to the function.

I have fixed this bug and pushed it.